### DISCUSSION TITLE
Install files in Common Documents

### DISCUSSION THREAD

**Discussion by amaitland** [oleg-shilo/wixsharp#1516] [Q&A]
Hi,

I have a requirement to install some files in the `Common Documents` folder (e.g. `C:\Users\Public\Documents`).

In .Net there's [Environment.SpecialFolder.CommonDocuments](https://learn.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=net-8.0) for this.

I'm trying to find the `Wix` equivalent. Some research suggests `WIX_DIR_COMMON_DOCUMENTS` is likely what I require. I'm not quite.

```
    <PropertyRef Id="WIX_DIR_COMMON_DOCUMENTS" />

    <Directory Id="TARGETDIR" Name="SourceDir">
      <Directory Id="WIX_DIR_COMMON_DOCUMENTS">`
        <Directory Id="MyFolder" Name="MyFolder" />`
      </Directory>`
    </Directory>`
```

Reference https://stackoverflow.com/a/17410330/4583726

Is there an easy way to map this to `WixSharp`? (I'm using `wix version 4`)

Appreciate any guidance!
Thanks,
Alex

**Comment by oleg-shilo** [comment]
Hey Alex, grittings from Melbourn.

Indeed the built-in constant for that folder is not defined in WiX. And this is the case when WixSharp cannot overcome this WiX limitation unless you use a custom UI.

The WiX workaround you are referring to can be done in WixSharp too.
Something along the lines:

```C#
 var project = new Project("Setup",
                   . . .
                   new PropertyRef("WIX_DIR_COMMON_DOCUMENTS"),
```
And to define the dir Id explicitly (by default WixSharp generates it based on the dir name), you can provide it as a constructor arg:

```C#
new Dir(new Id(WIX_DIR_COMMON_DOCUMENTS), "...
. . .
```
. . .

Unrelated...
Alex, great job on CefSharp. I used it and ... loved it :)

**Comment by amaitland** [comment]
> Hey Alex, grittings from Melbourn.

Hello from Brisbane! Thank you for the super quick reply! Greatly appreciated.

> The WiX workaround you are referring to can be done in WixSharp too.

Awesome, thank you for the guidance! 

> great job on CefSharp. I used it and ... loved it :)

I can say the same about `WixSharp` 😄 I had to redo a plain `wix v3` project recently, makes me appreciate just how awesome `WixSharp` is!

Thanks for all the hard work!
